wallets/src/main/java/bisq/wallets/WalletService.java

```diff
@@ -121,9 +160,21 @@ public boolean isWalletReady() {
         return CompletableFuture.supplyAsync(() -> {
             Wallet wallet = getWalletOrThrowException();
             double walletBalance = wallet.getBalance();
-            Coin coin = Coin.of(walletBalance, "BTC");
-            observableBalanceAsCoin.set(coin);
-            return coin.getValue();
+
+            // Balance changed?
+            if (observableBalanceAsCoin.get().getValue() != walletBalance) {
+                Coin coin = Coin.of(walletBalance, "BTC");
+                observableBalanceAsCoin.set(coin);
+
+                listUnspent().thenAccept(utxos -> {
+                    utxoTxIds.clear();
+                    utxos.stream()
+                            .map(Utxo::getTxId)
+                            .forEach(utxoTxIds::add);
+                });
+            }
+
+            return (long) walletBalance;
```

Comment:
Is walletBalance in satoshis? Weird why they use double then... support for sub-satoshis?

Reply:
It's in BTC.

Reply:
But then `return (long) walletBalance;` is wrong, no? As well as `observableBalanceAsCoin.get().getValue() != walletBalance`

Reply:
You're correct. I was distracted.